### PR TITLE
Parser: Fix @Ignored tests

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

- 3 ignored cases were old-parser bugs (one of which was quite interesting)
- 1 was a new-lexer bug, fixed here

See: https://www.pivotaltracker.com/story/show/183797809

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java [hunk 1]
```diff
@@ -950,15 +947,6 @@ public void testCaseWithComment() throws Exception {
...
 
-  @Test
-  @Ignore // Crashes old parser
-  public void testAlternationTypes() throws Exception {
-    parseTest("""
-    foo : [Integer | Text] -> (Integer | Text)
-    foo v = v.at 0
-    """);
-  }
-
```
Comment (kazcw) [inline engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java]: We don't need this test (we can't compare it against the old parser); it was just an interesting case

File: engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java [hunk 2]
```diff
@@ -1033,14 +1021,6 @@ public void testConstructorMultipleNamedArgs1() throws Exception {
...
   }
 
-  @Test
-  @Ignore // Old parser's representation of this is inconsistent with normal treatment of names.
-  public void testConstructorMultipleNamedArgs2() throws Exception {
-    parseTest("""
-    x = (Regex_Matcher.Regex_Matcher_Data case_sensitivity=Case_Sensitivity.Sensitive) dot_matches_newline=True
-    """);
-  }
-
```
Comment (kazcw) [inline engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java:1043]: Intentional difference from the old parser (the old parser would forget it was in type context in some situations, and use a representation of qualified names inconsistent with how it normally handles them in types)

File: engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java [hunk 3]
```diff
@@ -511,12 +511,9 @@ public void testRawBlockLiteral() throws Exception {
   }
 
   @Test
-  @Ignore
   public void testVariousKindsOfUnicodeWhitespace() throws Exception {
-    // mimics Text_Spec.enso:1049 and 722
-    // search for: # Disabled in the New Parser
     parseTest("""
-    t = '\\v\\f\\u{200a}\\u{202f}\\u{205F}\\u{3000}\\u{feff}'.trim
+    t = '\\v\\f\\u{200a}\\u{202f}\\u{205F}\\u{3000}'.trim
```
Comment (kazcw) [inline engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java:516]: Old parser was wrong; details in comments for `Text_Spec.enso` changes, below.

File: test/Tests/src/Data/Text_Spec.enso [hunk 4]
```diff
@@ -718,8 +718,7 @@ spec =
             'AB'.is_whitespace . should_be_false
             '  A   '.is_whitespace . should_be_false
 
-            # Disabled in the New Parser
-            # '\v\f\u{200a}\u{202f}\u{205F}\u{3000}\u{feff}'.is_whitespace . should_be_true
+            '\v\f\u{200a}\u{202f}\u{205F}\u{3000}'.is_whitespace . should_be_true
```
Comment (kazcw) [inline test/Tests/src/Data/Text_Spec.enso:729]: This turned out to be a fun case. `U+FEFF`, the Unicode [Byte Order Mark](https://en.wikipedia.org/wiki/Byte_order_mark), is not actually whitespace (Java doesn't consider it whitespace: https://www.fileformat.info/info/unicode/char/feff/index.htm; apparently whatever implements `.trim` doesn't classify it as such either). The original test should never have passed; the reason it did is somehow the old parser *loses* BOM characters--even if they originate from explicit escape sequences! I assume they are being stripped implicitly in some conversion between string representations.

Since the new parser doesn't lose this character on the way to producing IR, now we have to fix these tests to be consistent with the behavior of `trim`.

File: engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java [hunk 5]
```diff
@@ -1189,7 +1169,6 @@ public void testSimpleTrippleQuote() throws Exception {
   }
 
   @Test
-  @Ignore // enable CodeLocationsTest: "be correct in the presence of comments"
```
Comment (kazcw) [inline engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java:1192]: This was an edge case bug; I added a test case for this in `tests/parse.rs`, and fixed it in the lexer. I have a [refactor planned](https://www.pivotaltracker.com/story/show/183797744) to reduce the risk of bugs related to line-segmentation, which is currently more complicated than necessary.

**Review by jdunkerley — APPROVED** [review]

File: engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java (hunk 1 above)
Comment (hubertp) [reply engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java]: So the new one won't crash? Maybe still worth leaving then

**Review by hubertp — APPROVED** [review]
nit

**Review by JaroslavTulach — APPROVED** [review]
Thanks a lot for evaluation of the `@Ignore` tests.
